src/Adapter/Product/QueryHandler/GetRelatedProductsHandler.php

```diff
@@ -73,13 +73,17 @@ public function __construct(
      */
     public function handle(GetRelatedProducts $query): array
     {
-        $results = $this->productRepository->getRelatedProducts($query->getProductId(), $query->getLanguageId());
+        $productId = $query->getProductId();
+        $results = $this->productRepository->getRelatedProducts($productId, $query->getLanguageId());
+        // related products are not multishop compatible,
+        // so we just use default product shop to retrieve info required by multishop repositories
+        $shopId = $this->productRepository->getProductDefaultShopId($productId);
```

Comment:
Actually, each product has its own associations, so you should do this inside the loop for each different product ID.

Thus you don't need to store the productId inside a `$productId` especially since it creates confusion with the variable of the same name inside the loop